addons/web/controllers/main.py

```diff
@@ -1538,6 +1540,132 @@ def from_data(self, fields, rows):
         fp.close()
         return data
 
+
+class XMLExport(ExportFormat, http.Controller):
+    """Export records to XML
+
+    Export records in Odoo XML data files format. The exported fields of a model
+    are the union of all selected fields of that model.
+    """
+
+    @http.route('/web/export/xml', type='http', auth="user")
+    @serialize_exception
+    def index(self, data, token):
+        return self.base(data, token)
+
+    @property
+    def content_type(self):
+        return 'text/xml; charset=utf-8'
+
+    def filename(self, base):
+        return base + '.xml'
+
+    def extract_types(self, field_names, model):
+        """Extract models data for given fields
+
+        Update object types and comodels corresponding to a list of fields.
+
+        Args:
+            field_names: A list of fields.
+        """
+
+        models_types = defaultdict(dict)
+        models_comodels = defaultdict(dict)
+
+        for tokens in (field_name.split('/') for field_name in field_names):
+            cur_model = model
+            for tok in tokens:
+                if tok in ("id", ".id"):
+                    continue
+                if tok not in models_types[cur_model]:
+                    attrs = request.env[cur_model].fields_get([tok], attributes=['type', 'relation'])[tok]
+                    if attrs['type'] in ("one2many", "many2many", "many2one"):
+                        models_types[cur_model][tok] = attrs['type']
+                        models_comodels[cur_model][tok] = attrs['relation']
+                    else:
+                        models_types[cur_model][tok] = attrs['type']
+                if tok in models_comodels[cur_model]:
+                    cur_model = models_comodels[cur_model][tok]
+
+        self.models_types = models_types
+        self.models_comodels = models_comodels
```

Comment:
That's horrifying. Don't go randomly adding new parameters to singleton-ish objects, that's insane.